src/components/ReportActionItem/IssueCardMessage.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {useOnyx} from 'react-native-onyx';
+import Button from '@components/Button';
+import RenderHTML from '@components/RenderHTML';
+import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
+import useEnvironment from '@hooks/useEnvironment';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type {ReportAction} from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+
+type IssueCardMessageProps = {
+    action: OnyxEntry<ReportAction>;
+};
+
+function IssueCardMessage({action}: IssueCardMessageProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const {environmentURL} = useEnvironment();
+    // TODO: now mocking accountID with current user accountID instead of action.message.assigneeAccountID
+    const personalData = useCurrentUserPersonalDetails();
+    const [privatePersonalDetails] = useOnyx(ONYXKEYS.PRIVATE_PERSONAL_DETAILS);
+
+    // TODO: now mocking accountID with current user accountID instead of action.message.assigneeAccountID
+    const assignee = `<mention-user accountID=${personalData.accountID}></mention-user>`;
+    const link = `<a href='${environmentURL}/${ROUTES.SETTINGS_WALLET}'>${translate('cardPage.expensifyCard')}</a>`;
+
+    // TODO: remove || action?.originalMessage.html === 'NOMAILINGADDRESS' from the condition - added only for testing purposes, change privatePersonalDetails?.address for a proper address of user with assigneeAccountID
+    const noMailingAddress =
+        // @ts-expect-error 'action.originalMessage' is of type 'unknown'.
+        (action?.actionName === CONST.REPORT.ACTIONS.TYPE.CARD_MISSING_ADDRESS && isEmptyObject(privatePersonalDetails?.address)) || action?.originalMessage.html === 'NOMAILINGADDRESS';
+
+    const getTranslation = () => {
+        // switch (action?.actionName) {
+        // TODO: replace with previous line - added only for testing purposes
+        // @ts-expect-error 'action.originalMessage' is of type 'unknown'.
+        switch (action?.originalMessage?.html) {
+            case CONST.REPORT.ACTIONS.TYPE.CARD_ISSUED:
+                return translate('workspace.expensifyCard.issuedCard', assignee);
+            case CONST.REPORT.ACTIONS.TYPE.CARD_ISSUED_VIRTUAL:
+                return translate('workspace.expensifyCard.issuedCardVirtual', {assignee, link});
+            case CONST.REPORT.ACTIONS.TYPE.CARD_MISSING_ADDRESS:
+                return translate(`workspace.expensifyCard.${noMailingAddress ? 'issuedCardNoMailingAddress' : 'addedAddress'}`, assignee);
```

Comment:
That's because you have your address added - if you wouldn't have it a proper message with button to add an address would be displayed. After you add the address it is displayed like this - that's why I added two additional cases to test regardless you have it added or not

Reply:
You can try to log into a new account that doesn't have address added to test it

Reply:
how it's returning false?

Reply:
@allgandalf fixed! I forgot that I changed `action?.originalMessage.html ` for `action?.actionName` (the proper condition). I added an extra check that will be removed later